### PR TITLE
feat: Support setting CA certificate

### PR DESCRIPTION
**Pull request by vjda**
## Description

Support setting a custom CA certificate to validate SSL termination.

This change takes advantage of the ability of the `request` library to support setting custom CA certificates by passing the key `verify=/path/to/ca.pem` as an argument.

See [requests.request](https://requests.readthedocs.io/en/latest/api/#requests.request) for additional information.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How has it been tested ?

Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] New connection by providing a custom CA certificate path in key `verify=/path/to/ca.pem`
- [x] New connection by disabling SSL verification with `verify=False`
- [x] New connection by enabling SSL verification with `verify=True`

## Checklist

- [x] My PR is ready for prime time! Otherwise use the ["Draft PR" feature](https://help.github.com/en/articles/about-pull-requests#draft-pull-requests)
- [x] All commits have a correct title
- [x] Readme has been updated
- [x] Quality tests are green (see Codacy)
- [x] Automated tests are green (see pipeline)

### CODE REVIEW

File: pyartifactory/models/auth.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ class AuthModel(BaseModel):
 
     url: str
     auth: Tuple[str, SecretStr]
-    verify: bool = True
+    verify: bool | str = True
```
Comment (sigma67) [inline pyartifactory/models/auth.py:15]: this syntax does not work on Python < 3.9

Comment (vjda) [reply pyartifactory/models/auth.py]: Oh! my bad. It should be fixed by now.

File: README.md [hunk 2]
```diff
@@ -72,13 +73,22 @@ from pyartifactory import Artifactory
 art = Artifactory(url="ARTIFACTORY_URL", auth=('USERNAME','PASSORD_OR_API_KEY'), cert="/path_to_file/server.pem",api_version=1)
 ```
 
+Specify a local cert to use as custom CA certificate
+
+```python
+from pyartifactory import Artifactory
+art = Artifactory(url="ARTIFACTORY_URL", auth=('USERNAME','PASSORD_OR_API_KEY'), verify="/path_to_file/ca.pem", api_version=1)
```
Comment (anancarv) [inline README.md:80]: Why did you remove the 

The `verify` controls whether SSL/TLS certificate validation is performed for the server's certificate, while `cert` is used to provide a client-side certificate for mutual authentication when required by the server. These parameters are both related to SSL/TLS security but serve different purposes in making secure HTTP requests with the requests library.

They both must be documented. It is actually possible to set a [cert](https://github.com/anancarv/python-artifactory/blob/master/pyartifactory/models/auth.py#L16) key

**Review by anancarv — CHANGES_REQUESTED** [review]
You'll also need to add typing [here](https://github.com/anancarv/python-artifactory/blob/master/pyartifactory/objects.py#L76)

**Comment by vjda** [comment]
> You'll also need to add typing [here](https://github.com/anancarv/python-artifactory/blob/master/pyartifactory/objects.py#L76)

Fixed

File: README.md (hunk 2 above)
Comment (vjda) [reply README.md]: Done. 

@anancarv, @sigma67 please let me know if everything is now ok for you. Cheers!

Comment (anancarv) [reply README.md]: LGTM, you'll only need to resolve the conflicts and it's gonna be good to be merged

Comment (vjda) [reply README.md]: Already done!

**Review by anancarv — DISMISSED** [review]
lgtm

**Review by anancarv — APPROVED** [review]

**Comment by vjda** [comment]
It seems that test matrix stages are failing in the Github workflow. However I ran the tests on my laptop and they worked OK.

Checking the commit history I think #144 broke the code coverage reporting .

```
2023-10-09 17:30:07.714Z error [CodacyCoverageReporter] Invalid configuration: Empty argument for --project-token  - (CodacyCoverageReporter.scala:25)
```

Not quite sure how to fix it but happy to help if you want 😀

**Comment by anancarv** [comment]
> It seems that test matrix stages are failing in the Github workflow. However I ran the tests on my laptop and they worked OK.
> 
> Checking the commit history I think #144 broke the code coverage reporting .
> 
> ```
> 2023-10-09 17:30:07.714Z error [CodacyCoverageReporter] Invalid configuration: Empty argument for --project-token  - (CodacyCoverageReporter.scala:25)
> ```
> 
> Not quite sure how to fix it but happy to help if you want 😀

Don't worry about that. It's because you don't have the codacy token set on your repo.

**Pull request merged**